tests/phpunit/src/BltProjectTestBase.php

```diff
@@ -57,10 +57,24 @@ abstract class BltProjectTestBase extends TestCase {
   protected $site2Dir;
   protected $sandboxInstanceClone;
 
+  /**
+   * @var bool
+   *
+   * Track whether our master sandbox has been initialized.
+   */
+  protected static $initialized = FALSE;
+
   /**
    * {@inheritdoc}
+   * @throws \Exception
    */
   public function setUp() {
+    if (!self::$initialized) {
```

Comment:
Since you don't need anything on the instance, why not put this in the static `setUpBeforeClass()` which is called once before the first test?

Reply:
The problem with that is that BltProjectTestBase is never invoked directly, it's only a base class for all of our other test cases. _Each time_ one of those classes is run, it will also run setUpBeforeClass(), which is a huge waste of resources. It only needs to run once for the entire suite.

This is the only way I know of to ensure this only runs once for the entire suite instead of once per class.

Reply:
That makes sense. Still, if you put the same solution in setUpBeforeClass it will be less calls to this method, but that's nitpicking now.

You could put this in a static method so that its obvious why, as you said in this comment, e.g. `bootstrapSandboxOncePerSuite()`

Reply:
Yep you are right. Thanks for being persistent :) I swear I tried using setupBeforeClass originally and it didn't work for some reason, but I just tried it again and it worked great :man_shrugging: I'll push a change.